src/flb_async_timer.c

```diff
@@ -21,103 +21,107 @@
 #include <fluent-bit/flb_thread_pool.h>
 #include <fluent-bit/flb_output_thread.h>
 #include <fluent-bit/flb_async_timer.h>
+#include <fluent-bit/flb_scheduler.h>
 
-void flb_async_timer_destroy(struct flb_out_async_timer *timer)
+void flb_async_timer_destroy(struct flb_async_timer *timer)
 {
     mk_list_del(&timer->_head);
     flb_coro_destroy(timer->coro);
     flb_free(timer);
 }
 
-void flb_async_timer_cleanup(struct mk_list *destroy_list)
+void flb_async_timer_cleanup(struct flb_sched *sched)
 {
-    struct flb_out_async_timer *async_timer;
+    struct flb_async_timer *async_timer;
     struct mk_list *tmp;
     struct mk_list *head;
+    struct mk_list *destroy_list = &sched->async_timer_list_destroy;
+
     mk_list_foreach_safe(head, tmp, destroy_list) {
-        async_timer = mk_list_entry(head, struct flb_out_async_timer, _head);
+        async_timer = mk_list_entry(head, struct flb_async_timer, _head);
```

Comment:
TODO: need to acquire mutex every time the engine reads or the worker thread writes/deletes.